applications/photo-asset-manager/README_TEMPLATE.md

```diff
@@ -0,0 +1,205 @@
+| LabelsFn | APIG GET /labels | N/A | {"labels": {"maintain": {"count": 5}} | Storage bucket, Label table |  
+| | | | | |
+
+**Note**: The application logic required to build these AWS Lambda functions is discussed later in this document.
+
+**Upload**
+
+The following JSON represents the input for the **UploadFn** Lambda function.
+
+```xml
+{
+  "body": "{\"file_name\":\"mountain.jpg\"}",
+  "resource": "/upload",
+  "path": "/upload",
+  "httpMethod": "PUT"
+}
+
+```
+
+You need to use this JSON if you want to test your AWS Lambda function by using the AWS Management Console. You specify the JSON in the **Event JSON** section, as shown in the following illustration.
+
+![AWS Photo Analyzer](images/EventJSON.png)
+
+This Lambda function returns JSON that contains the presigned URL that can be used to upload the .jpg file specified in the URL (in this example, mountain.jpg).
+
+```xml
+{
+  "statusCode": 200,
+  "headers": {
+    "Access-Control-Allow-Origin": "*"
+  },
+  "body": "{\"url\":\"https://xxxxxxxxxxxx.amazonaws.com/xxxxxxxxxxxxxxx-mountain.jpg?X-Amz-Security-xxxxxxxxxxxxxxxxxxx\"}",
+  "isBase64Encoded": false
+ }
+```
+
+**LabelsFn**
+
+The following JSON represents the input for the **LabelsFn** Lambda function.
+
+```xml
+  {
+   "tags": "{}",
+   "resource": "/labels",
+   "path": "/labels",
+   "httpMethod": "GET"
+  }
+
+```
+
+#### API Gateway
+
+API Gateway provides HTTP API routes for the **UploadFn**, **LabelsFn**, **Labels**, and **DownloadFn** AWS Lambda functions. Parameters for all routes are provided in the body of the request in a JSON object. The following table describes the API Gateway routes.
+| Method | Route | Parameters | Example response | Lambda |
+| ------------- | -----------------| --------------------------| --------------------------------------------------------------- | ----------------------------|
+| PUT | /upload | filename: string | {"url": "presigned URL"} | UploadFn |
+| GET | /labels | N/A | {"labels": {"maintain": {"count": 5}, "lake": {"count": 3}}} | LabelsFn |
+| POST | /download | labels: string[] | {} (event) | DownloadFn |
+
+#### S3 buckets
+
+The PAM application uses two S3 buckets:
+
+- **{NAME}-sdk-code-examples-pam-storage-bucket** - Used as the storage bucket.
+- **{NAME}-sdk-code-examples-pam-working-bucket** - Used as the working bucket and provides zip download storage.
+
+**Note**: The name value is defined when you run the AWS CDK script to set up the resources. This is discussed later in this document.
+
+#### Amazon DynamoDB
+
+The PAM application uses a DynamoDB table to track data. The Labels table named **{NAME}-SDKCodeExamplesPAM-Labels** contains the labels found by Amazon Rekognition. It has a simple primary key named **Label** with an attribute Label of type S.
+**Note**: This Amazon DynamoDB table is created when you run the AWS CDK script to set up the resources. This is discussed later in this document.
```

Comment:
Same question as above on "discussed in this document"?

Reply:
Same answer